prometheus_hardware_exporter/collector.py

```diff
@@ -109,3 +110,293 @@ def fetch(self) -> List[Payload]:
+
+    sas2ircu = Sasircu(2)
+
+    @property
+    def specifications(self) -> List[Specification]:
+        """Define LSI SAS-2 metric specs."""
+        return [
+            Specification(
+                name="lsi_sas_2_controllers",
+                documentation="Number of controllers",
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_ir_volumes",
+                documentation="Number of integrated RAID volumes",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_ir_volume",  # will append "_info" internally
+                documentation="Shows the information about the integrated RAID volume",
+                metric_class=InfoMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_physical_devices",
+                documentation="Number of physical devices",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_physical_device",  # will append "_info" internally
+                documentation="Shows the information about the physical device",
+                metric_class=InfoMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_enclosure",  # will append "_info" internally
+                documentation="Show the information about the enclosure",
+                metric_class=InfoMetricFamily,
+            ),
+            Specification(
+                name="sas2ircu_command_success",
+                documentation="Indicates if the command is successful or not",
+                metric_class=GaugeMetricFamily,
+            ),
+        ]
+
+    def fetch(self) -> List[Payload]:
+        """Load the LSI SAS-2 controllers related information."""
+        adapters = self.sas2ircu.get_adapters()
+        all_information = [(idx, self.sas2ircu.get_all_information(idx)) for idx in adapters]
+
+        if not all([adapters, all_information]):
+            logger.error(
+                "Failed to get LSI SAS-2 controller information using %s", self.sas2ircu.command
+            )
+            return [
+                Payload(
+                    name="sas2ircu_command_success",
+                    value=0.0,
+                )
+            ]
+
+        payloads = [
+            Payload(
+                name="lsi_sas_2_controllers",
+                value=len(adapters),
+            ),
+            Payload(
+                name="sas2ircu_command_success",
+                value=1.0,
+            ),
+        ]
+
+        for idx, info in all_information:
+            # Add integrated RAID volume metrics
+            if info["ir_volumes"]:
+                payloads.append(
+                    Payload(
+                        name="lsi_sas_2_ir_volumes",
```

Comment:
The suffix of `_count` / `_total` will suggest that it is a Counter metric, and a counter metric can only increase or reset to zero. This is not the case for us. On a second thought, if this metric is really causing trouble for you, I will suggest that we remove it entirely. Since the same information is contained in `lsi_sas_3_ir_volume_info` metric (i.e. we can build a volume count metric out of it, making use of the uniqueness of the volume_id). Does it work for you?